Debugging Fundamentals/Task1/Product.cs

```diff
@@ -11,5 +11,22 @@ public Product(string name, double price)
         public string Name { get; set; }
 
         public double Price { get; set; }
+
+        public override bool Equals(object obj)
```

Comment:
Could you please review and update your logic of Equals override?
https://learn.microsoft.com/en-us/previous-versions/ms173147(v=vs.90)?redirectedfrom=MSDN

Reply:
I followed the guideline and implemented two Equals method, one for System.Object type and one for Product type as recommended.

Reply:
Sorry I have send you outdated link, you need follow this one: https://learn.microsoft.com/en-us/visualstudio/ide/reference/generate-equals-gethashcode-methods?view=vs-2022
In your current implementation you have dublicated loginc for two Equals. But one of them should reuse logic from the second one

Reply:
Oh, I see. I followed the new document and liked this approach more 😄